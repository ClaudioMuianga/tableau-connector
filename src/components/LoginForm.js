import React, { Component } from 'react'
import {
  Button,
  Col,  
  Grid,
  Row
} from 'react-bootstrap'

import analytics from '../analytics'
import './LoginForm.css'
import sparkle from '../static/img/new-sparkle-logo.png'

class LoginForm extends Component {

  constructor () {
    super()
    this.oauthClientId = process.env.REACT_APP_OAUTH_CLIENT_ID
    this.oauthRedirectURI = process.env.REACT_APP_OAUTH_REDIRECT_URI
  }

  authenticateButtonClicked = () => {
    analytics.track('tableauconnector.auth_form.auth_button.click')
  }

  render () {
    return (
      <Grid className='loginForm'>
        <Row className='center-block'>
          <Col md={10} mdOffset={1} xs={10} xsOffset={1}>
            <img src={sparkle} className='center-block' alt='data.world sparkle logo' />
            <h3 className='header'>Authenticate with data.world to access your dataset</h3>
            <div className='text-center'>
              <Button
                onClick={this.authenticateButtonClicked}
                href={`https://data.world/oauth/authorize?client_id=${this.oauthClientId}&redirect_uri=${this.oauthRedirectURI}`}
                bsStyle='primary'>Authenticate</Button>
            </div>
          </Col>
        </Row>
      </Grid>
    )
  }
}

export default LoginForm