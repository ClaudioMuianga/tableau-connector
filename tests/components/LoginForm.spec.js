import '../util'
import renderer from 'react-test-renderer'

import React from 'react'
import ReactDOM from 'react-dom'
import LoginForm from '../../src/components/LoginForm'

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<LoginForm />, div)
})

it('renders form', () => {
  expect(renderer.create(<LoginForm />).toJSON()).toMatchSnapshot()
})