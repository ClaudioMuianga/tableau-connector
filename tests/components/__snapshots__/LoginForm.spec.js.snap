exports[`test renders form 1`] = `
<div
  className="loginForm container">
  <div
    className="center-block row">
    <div
      className="col-md-10 col-md-offset-1 col-xs-10 col-xs-offset-1">
      <img
        alt="data.world sparkle logo"
        className="center-block"
        src="new-sparkle-logo.png" />
      <h3
        className="header">
        Authenticate with data.world to access your dataset
      </h3>
      <div
        className="text-center">
        <a
          className="btn btn-primary"
          href="https://data.world/oauth/authorize?client_id=data-dot-world-tableau&redirect_uri=https://dw-tableau-oauth.herokuapp.com/callback"
          onClick={[Function]}>
          Authenticate
        </a>
      </div>
    </div>
  </div>
</div>
`;
